internal/proto/internal.proto

```diff
@@ -67,4 +67,48 @@ message PrContents {
 
   minder.v1.PullRequest pr = 1;
   repeated File files = 2;
-}
\ No newline at end of file
+}
+
+message SelectorProvider {
+  // the name of the provider, e.g. github-app-jakubtestorg
+  string name = 1;
+  // the class of the provider, e.g. github-app
+  string class = 2;
+}
+
+message SelectorRepository {
+  // the full name of the repository, e.g. stacklok/minder
+  string name = 1;
+  // the provider of the repository
+  SelectorProvider provider = 2;
+
+  // is_fork is true if the repository is a fork, nil if "don't know" or rather
+  // not applicable to this provider
+  optional bool is_fork = 3;
+  // is_private is true if the repository is private, nil if "don't know" or rather
+  // not applicable to this provider
+  optional bool is_private = 4;
+}
+
+message SelectorArtifact {
+  // the full name of the artifact, e.g. stacklok/minder-server
+  string name = 1;
+  // the provider of the artifact
+  SelectorProvider provider = 2;
+
+  // the type of the artifact, e.g. "container"
+  string type = 4;
+}
+
+message SelectorEntity {
+  // one of repository, pull_request, artifact (see oneof entity)
+  minder.v1.Entity entity_type = 1;
+  // the name of the entity, same as the name in the entity message
+  string name = 2;
```

Comment:
This is to allow for access as both `name` and `repository.name`?

Reply:
Maybe I should have spelled this out more prominently in the design doc. There's some examples in the [unit tests](https://github.com/jhrozek/minder/blob/35e207a748d5e1239c9e1cbee2c917e2476778cb/internal/engine/selectors/selectors_test.go#L150)

and here's a simple inline example. Given these selectors:
```
  - comment: This should not apply to bad-go
    entity: repository
    selector: repository.name != 'jakubtestorg/bad-go'
  - comment: Skip any entity that contains demo in the name
    selector: entity.name.contains('demo') == false
```
The first one applies to repositories only, in this case the selector uses `repository.name`, but it could use also `repository.provider.name/class`, `repository.is_fork` or `repository.is_private`.

The second one applies to any entity type. It can use only attributes from the `SelectorEntity` message which currently are `entity.name`, `entity.provider.name` and `entity.provider.class`. (Let's talk about adding the purl or global name separately).

In both the entity-generic message and the entity-specific message we'll also add the k-v map.